pytest_mh/utils/firewall.py

```diff
@@ -289,95 +302,77 @@ def _next_priority(self) -> int:
         self._priority -= 1
         return priority
 
-    def add_direct_rule(
+    def add_policy(
         self,
-        chain: str,
-        args: list[Any],
+        name: str,
         *,
-        table: str = "filter",
-        ip_family: Literal["ipv4", "ipv6", "all"] = "all",
+        ingress: str | None = None,
+        egress: str | None = None,
         priority: int | None = None,
+        target: Literal["CONTINUE", "ACCEPT", "DROP", "REJECT"] | None = None,
     ) -> int:
         """
-        Add a new direct rule.
-
-        This methods returns a priority of this rule. You need to use this
-        priority if you remove the rule with :meth:`remove_direct_rule`.
-
-        :param chain: iptables chain (e.g. INPUT or OUTPUT).
-        :type chain: str
-        :param args: iptables arguments
-        :type args: list[Any]
-        :param table: iptables table, defaults to "filter"
-        :type table: str, optional
-        :param ip_family: If the rules is added as IPv4, IPv6 rule or both, defaults to ``all``.
-        :type ip_family: Literal["ipv4", "ipv6", "all"], optional
-        :param priority: Rule priority, defaults to None (= auto-assign next value)
-        :type priority: int | None, optional
-        :return: Rule priority, to be used for rule removal.
+        Add a new (permanent) policy.
+
+        Except for the name, all parameters are optional. When the priority is not provided,
+        the next priority is assigned. When the other parameters is not provided,
+        no value is assigned to the newly created policy and it defaults to ``firewalld``'s
+        defaults.
+
+        :param ingress: The ingress zone, not assigned if not provided.
+        :type ingress: str | None, optional.
+        :param egress: The egress zone, not assigned if not provided.
+        :type egress: str | None, optional.
+        :param priority: Rule priority, defaults to the next priority.
+        :type priority: int | None, optional.
+        :param target: Rule target, not assigned if not provided.
+        :type target: Literal["CONTINUE", "ACCEPT", "DROP", "REJECT"] | None, optional.
+        :return: Policy priority.
         :rtype: int
         """
+        self.logger.info(f'Firewalld: adding policy "{name}"')
+
+        cmd = ["firewall-cmd", "--permanent", "--new-policy", name]
+        self.host.ssh.exec(cmd, log_level=SSHLog.Error)
+        cmd[2] = "--policy"
+
         if priority is None:
             priority = self._next_priority
```

Comment:
I wonder, shouldn't this variable change at some point?

Reply:
It's not a variable but a property. It changes every time it is used. Check line 290.

Reply:
Now I see the `@property` attribute. Thanks!